impl/src/main/java/org/jboss/weld/util/Beans.java

```diff
@@ -647,6 +648,15 @@ public static boolean hasBuiltinScope(Bean<?> bean) {
                 || ConversationScoped.class.equals(bean.getScope()) || Dependent.class.equals(bean.getScope());
     }
 
+    public static Class<? extends Annotation> getBeanDefiningMetaAnnotation(AnnotatedType<?> annotatedType) {
```

Comment:
I think it should be rather something like `getBeanDefiningAnnotationScope()`. This method does not return a meta annotation (e.g. `NormalScope`) but the scope types which are bean defining annotations. 

Otherwise looks perfect!